src/core/logging.js

```diff
@@ -1,27 +1,42 @@
 import config from "./config";
 import { objectType } from "./utilities";
 import Promise from "../promise";
+import {
+	setTimeout
+} from "../globals";
+
+function _promisfyCallbacksSequentially( callbacks, args ) {
+	return callbacks.reduce( ( promiseChain, callback ) => {
+		const executeCallback = () => callback( args );
+		return promiseChain.then(
+			executeCallback,
+			( err ) => {
+				setTimeout( executeCallback );
```

Comment:
Use of `setTimeout` must be conditional for non-browser environments such as SpiderMonkey (see other references). An alternate approach here can be to use `Promise.resolve()` as fallback for `setTimeout`, or to avoid this alltogether by using `Promise.finally()` for the executeCallback, which naturally forwards the original rejection/error.